runtime/buffer.cc

```diff
@@ -478,79 +478,96 @@ SLINKY_NO_INLINE index_t make_for_each_loops_impl(
   void* plan = plan_base;
   index_t slice_extent = 1;
   index_t extent = 1;
-  for (index_t d = static_cast<index_t>(buf->rank) - 1; d >= 0; --d) {
-    const dim& buf_dim = buf->dim(d);
-
-    if (buf_dim.min() == buf_dim.max()) {
-      // extent 1, we don't need any of the logic here, skip to below.
-    } else if (buf_dim.max() > buf_dim.min()) {
-      if (use_folded_loop(bufs, bufs_size, d)) {
-        // extent > 1 and there is a folded dimension in one of the buffers, or we need to crop one of the buffers.
-        assert(extent == 1);
-        for_each_loop* loop = increment_plan<for_each_loop>(plan);
-        loop->impl = for_each_loop::folded;
-        loop->extent = buf_dim.extent();
-        prev_loop = loop;
-
-        const dim** dims = increment_plan<const dim*>(plan, bufs_size);
-        dims[0] = &buf->dim(d);
-        for (std::size_t n = 1; n < bufs_size; n++) {
-          dims[n] = d < static_cast<index_t>(bufs[n]->rank) ? &bufs[n]->dim(d) : &broadcast_dim;
+  if (buf->rank == 0) {
```

Comment:
I have put an insane amount of time into this small bit of code to make sure it is as fast as possible, so I'd rather not change it just for the sake of sanitizers, a new branch is a big deal that could change a lot.

Since this is just a sanitizer fix, we should just put an `#ifdef UNDEFINED_BEHAVIOR_SANITIZER`, and only change control flow in that block. Just return early in the case `buf->rank == 0` so you don't need to change the whole function.

Reply:
Leaving known UB here rubs me the wrong way, but I'll do as you suggest